### PR TITLE
Document Viewport API

### PR DESCRIPTION
**Pull request by macdrevx**
## Pull request checklist:
 - [x] Add documentation comments for any added or updated public APIs.
 - [x] Review and agree to the Contributor License Agreement ([CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement)).

### CODE REVIEW

File: Sources/MapboxMaps/Viewport/Transitions/Default/DefaultViewportTransition.swift [hunk 1]
```diff
@@ -1,7 +1,13 @@
+/// A default ``ViewportTransition`` implementation.
+///
+/// Use ``Viewport/makeDefaultViewportTransition(options:)`` to create instances of this
+/// class.
```
Comment (pengdev) [inline Sources/MapboxMaps/Viewport/Transitions/Default/DefaultViewportTransition.swift:4]: should we mention user should keep a reference to the states and transitions for future operations?

File: Sources/MapboxMaps/Viewport/ViewportState.swift [hunk 2]
```diff
@@ -1,27 +1,64 @@
...
-// without requiring that they conform to Hashable. One implication is that they
-// must not be shared among multiple MapView/Viewport instances.
-//
-// States should generally pre-warm their data sources as soon as they are created
-// to minimize delays when they become current. For this reason, only states that
-// are currently needed should be kept alive.
+/// `ViewportState` is a protocol that ``Viewport`` depends on as it orchestrates transitions to and
+/// from different states.
+///
+/// A `ViewportState` is a reference type and must not be shared among multiple ``Viewport``
```
Comment (pengdev) [inline Sources/MapboxMaps/Viewport/ViewportState.swift:4]: I think we don't support multiple viewport instances?

File: Sources/MapboxMaps/Foundation/MapView.swift [hunk 3]
```diff
@@ -40,6 +40,10 @@ open class MapView: UIView {
     /// Manages the configuration of custom view annotations on the map.
     public private(set) var viewAnnotations: ViewAnnotationManager!
 
+    /// ``Viewport`` is a high-level and extensible API for driving the map camera. It
+    /// provides built-in states for following the location puck and showing an overview of
```
[the remark this answers is not in the document]
Comment (macdrevx) [reply Sources/MapboxMaps/Foundation/MapView.swift:44]: Updating.

File: Sources/MapboxMaps/Viewport/Transitions/Default/DefaultViewportTransition.swift (hunk 1 above)
Comment (macdrevx) [reply Sources/MapboxMaps/Viewport/Transitions/Default/DefaultViewportTransition.swift:4]: I'll mention this in the docs for `transition(to:…)` and `defaultTransition`

**Pull request merged**